pkg/reconciler/instances/rafter/action_test.go

```diff
@@ -0,0 +1,188 @@
+package rafter
+
+import (
+	"context"
+	"testing"
+
+	log "github.com/kyma-incubator/reconciler/pkg/logger"
+
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes/mocks"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/workspace"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	fake "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestEnsureRafterSecret(t *testing.T) {
+
+	// test cases
+	tests := []struct {
+		Name            string
+		Values          *rafterValues
+		PreCreateSecret bool
+		ExpectSecret    bool
+	}{
+		{
```

Comment:
We can add use test for `randAlphaNum` generating access key and key and check if size of string is ok.